omniledger/service/service.go

```diff
@@ -1158,7 +1170,27 @@ func (s *Service) save() {
 	defer s.storage.Unlock()
 	err := s.Save(storageID, s.storage)
 	if err != nil {
-		log.Error("Couldn't save file:", err)
+		log.Error(s.ServerIdentity(), "Couldn't save file:", err)
+	}
+}
+
+func (s *Service) trySyncAll() {
+	gas := &skipchain.GetAllSkipChainIDs{}
+	gasr, err := s.skService().GetAllSkipChainIDs(gas)
+	if err != nil {
+		log.Error(s.ServerIdentity(), err)
+		return
+	}
+	for _, scID := range gasr.IDs {
+		sb, err := s.db().GetLatestByID(scID)
+		if err != nil {
+			log.Error(s.ServerIdentity(), err)
+			continue
+		}
+		err = s.skService().SyncChain(sb.Roster, sb.Hash)
```

Comment:
Yes, it will start and stop go-routines. That was the intended design because I didn't want to distinguish between the case where blocks arrive normally and the case where lots of blocks arrive as a result of a synchronisation. Hence, node that are synchronising might send heartbeat messages that don't make sense for the online nodes. The online nodes should continue to behave correctly because they know the message came from a wrong leader. We don't have an explicit test for it but it might be good to have one in the future. Implicitly, as long as synchronisation doesn't break online nodes, which it doesn't (and tested), then I think we're in a good shape.